### ISSUE TITLE
Preceeding align prevents create/modify function 

### ISSUE THREAD

**Issue by tamlin-mike** [NationalSecurityAgency/ghidra#5493]
Error message: "Function entryPoint may not be created on defined data"

- A function having a preceeding align (Builtin/Alignment) can no longer be modified.

- Any code having a preceeding align prevents defining function.

Win32 PE.
1. Find some code with 2 or more `cc` padding at the end.
2. You want to get rid of the display of that junk, so you select the `cc` bytes, and create an "alignment" (1) from them.
3. Place cursor on the byte/function/label following the now created align section, and press `F`.
4. Experience the error message.

**Expected behavior**
Obvious.

- OS: Windows, Java 18.0.2.1. Ghidra 10.3.1 (public build), from GitHub distro.

(1) It's not even an alignment, it's a _filler_. The number it displays inside the parsens is number of _fill_ bytes. An actual (correct) alignment would hold a value 2^n, telling what it's aligning the following address to.

**Comment by ghidra1** [comment]
You are correct, the `AlignmentDataType` should simply be called padding/filler since its length in bytes is exactly what it was created with.    I am unable to reproduce the problem as you describe.  I also see no reason why the issue should occur.  I also added defined data after a fallthrough instruction and still was unable to reproduce.  You may need to provide us with an exported gzf file with a specified address where we can click label (w/Alignment data already in-place) and hit F-action to reproduce failure.

**Comment by tamlin-mike** [comment]
I'm starting to wonder if I got bitten by the "persistent selection" (not knowing the actual name of the "feature", or even why it exists). It might have been that I had previously selected some text (using keyboard, holding shift+cursor movement keys) to copy it, or maybe even the selection (the `cc` bytes - Alignment failed to auto-deduce the block end, why I had to select the whole block before invoking Alignment) was still in a "persistent selection" state. If that's the source of my problem, is there some way to disable that "feature", or at least get out of that "mode" (i.e. cancel/remove selection)? Esc doesn't work, and that's when I gave up trying to find a keyboard way.

I have since found out a workaround is to use the _mouse_ to click somewhere else in the text view to remove the "persistent selection", but being forced to use the mouse to disable a selection made using the keyboard seems... illogical.

Since you wrote "where we can **click** label", was that what you did when trying to reproduce? I most likely used only keyboard to navigate to the position in the Listing view. IFF that was my original problem. since mouse click has the side-effect of removing/disabling the "persistent selection", you could likely never reproduce the problem.



While on the subject of keyboard navigation/edit (let me know if I should open a separate issue for this), it would be nice if platform standard editing keys were working. e.g. on Windows the "noob" keys for cut/copy/paste are Ctrl+X, Ctrl+C, Ctrl+V, but the keyboard shortcut for more seasoned keyboard-warriors that have also always (since at least three decades+) been Ctrl+Del, Ctrl+Ins, Shift+Ins does not work in Ghidra. That lack of this basic functionality is not only very irritating, it has already bitten me hard many times, leaving stale data on the clipboard, leading to great confusion, mis-navigation and some gnashing of teeth.

**Comment by ghidra1** [comment]
> Since you wrote "where we can **click** label", was that what you did when trying to reproduce?

I tried this, however, create function is based on the current address/selection.  In my attempt I did not have a selection.  If there is a slection, that will be used as the function body with the minimum address being the entry point.

@tamlin-mike  Can we close this ticket?

**Comment by ghidra1** [comment]
Yes, selection takes precedence over location for all actions/plugins which seupport the use of a selection.  This includes function creation.  Without a selection the current location is use as the function entry and the body computed based instruction flow.  With a selection the body is forced to equal the selection (with or without instructions) and min address is made entry point.
